### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by w8jcik**
* [x] Adds CMake configuration
* [x] Alters README.md
* [x] Centers application on the screen

To test it
```
git clone https://github.com/emscripten-core/emsdk.git
(cd emsdk && ./emsdk install latest && ./emsdk activate latest)
source emsdk/emsdk_env.sh

git clone --recursive --branch add/cmake https://github.com/w8jcik/lv_sim_emscripten.git
cd lv_sim_emscripten
emconfigure cmake -DCMAKE_TOOLCHAIN_FILE="${EMSDK}/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake" -S . -B build && cmake --build build -j $(nproc)
```

* Open `about:config` in Firefox.
* Find `privacy.file_unique_origin` and change it to `false`.
* Close Firefox.

```
firefox build/lvgl.html
```

### CODE REVIEW

**Comment by embeddedt** [comment]
I'm not sure about having multiple build systems available within one project. It seems like it would make maintenance harder. On the other hand, we do have both a `Makefile` and a CMake file for the regular PC simulator.

@kisvegabor What are your thoughts?

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.12)
+project (lvgl)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O2 -s USE_SDL=2")
+
+include_directories(${PROJECT_SOURCE_DIR})
+
+file(GLOB INCLUDES1 "./*.h")
+file(GLOB_RECURSE INCLUDES2 "lv_drivers/*.h" "lv_examples/*.c" "lvgl/*.h")
```
Comment (kisvegabor) [inline CMakeLists.txt]: Should be "lv_examples/*.h", isn't it?

**Comment by kisvegabor** [comment]
IMO as CMakeLists.txt has very few hardcoded paths and file names the maintenance is not an issue. I suppose it will need very minimal maintenance as we have these 3 folders for a very long time. 

In addition, while it's easy to use Makefile on Linux, CMake is much easier on Window (correct me if I wrong here, I don't really have development experiment on Windows)

File: CMakeLists.txt (hunk 1 above)
Comment (w8jcik) [reply CMakeLists.txt]: Fixed

**Comment by w8jcik** [comment]
We can also replace
```cmake
file(GLOB INCLUDES1 "./*.h")
file(GLOB_RECURSE INCLUDES2 "lv_drivers/*.h" "lv_examples/*.h" "lvgl/*.h")

file(GLOB SOURCES1  "./*.c")
file(GLOB_RECURSE SOURCES2  "lv_drivers/*.c" "lv_examples/*.c" "lvgl/*.c")

set(INCLUDES ${INCLUDES1} ${INCLUDES2})
set(SOURCES ${SOURCES1} ${SOURCES2})
```
with
```cmake
file(GLOB_RECURSE INCLUDES "./*.h")
file(GLOB_RECURSE SOURCES "./*.c")
```

Then there are no hardcoded pahts. I did it this way to make it easier for user to disable one of the folders or files.

**Comment by kisvegabor** [comment]
It'd be even better regarding maintenance. 
Please update it as you suggested.

**Comment by w8jcik** [comment]
Done

File: lvgl_shell.html [hunk 2]
```diff
@@ -1,4 +1,18 @@
 <html>
+	<head>
+		<style type="text/css">
+			html, body {
+			  margin: 0;
+			}
+
+			body {
+			  height: 100%;
```
Comment (embeddedt) [inline lvgl_shell.html]: In my experience, you need full width and height on both `body` and `html` for it to actually center on all browsers.

```css
html, body {
    width: 100%;
    height: 100%;
    min-width: 100%;
    min-height: 100%;
}
```

Comment (w8jcik) [reply lvgl_shell.html]: Done

**Comment by embeddedt** [comment]
Looks good! Thanks for the contribution!

**Pull request merged**